pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -267,23 +269,84 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		reqLogger.Info("Created Service", "Service", serDNS)
 
 		if infinispan.IsExposed() {
-			externalService := r.serviceExternal(infinispan)
-			if externalService != nil {
-				err = r.client.Create(context.TODO(), externalService)
-				if err != nil && !errors.IsAlreadyExists(err) {
-					reqLogger.Error(err, "failed to create external Service", "Service", ser)
+			switch infinispan.Spec.Expose.Type {
+			case infinispanv1.ExposeTypeLoadBalancer, infinispanv1.ExposeTypeNodePort:
+				externalService := r.serviceExternal(infinispan)
+				if externalService != nil {
```

Comment:
@rigazilla Looks like serviceExternal can return nil not case of Unsupported type as it doesn't have any error handlers and mainly in case of problem inside code.

Reply:
yeah, serviceExternal just populate a struct. I've cleanup the if statement